docs/NetworkPeeringForScalarDLAuditor.md

```diff
@@ -0,0 +1,62 @@
+# Create network peering for Scalar DL Auditor mode
+
+This document explains how to connect several private networks for ScalarDL Auditor mode. To make ScalarDL Auditor mode work properly, ScalarDL Ledger and ScalarDL Auditor need to connect with each other. 
+
+## Private network you must connect
+
+Depending on your application (client) deployment, you must connect several private networks.
+
+### Connect two private networks
+
+If you deploy your application on the same private network as ScalarDL Ledger or ScalarDL Auditor, you must connect two private networks.
+
+* [ScalarDL Ledger network] <-> [ScalarDL Auditor network]
+
+### Connect three private networks
+
+If you deploy your application (client) on another private network than ScalarDL Ledger or ScalarDL Auditor, you must connect three private networks.
+
+* [ScalarDL Ledger network] <-> [ScalarDL Auditor network]
+* [ScalarDL Ledger network] <-> [Application (client) network]
+* [ScalarDL Auditor network] <-> [Application (client) network]
+
+## Network requirements
+
+### IP address ranges
+
+To avoid conflicting IP addresses between several private networks, you must create several private networks with different IP address ranges.
+
+* Example
+    * Private network for ScalarDL Ledger -> 10.1.0.0/16
+    * Private network for ScalarDL Auditor -> 10.2.0.0/16
+    * Private network for application (client) -> 10.3.0.0/16
+
+### Connections
+
+The connections (ports) that are used between ScalarDL Ledger, ScalarDL Auditor, and the client by default are the following. You must allow these connections between each private network. Note that if you change the listening port of ScalarDL in a configuration file ([ledger|auditor].properties) from default, you must allow the connections using the port you configured.
+
+* Scalar DL Ledger
+    * 50051/TCP (Accept the requests from a client and Auditor)
+    * 50052/TCP (Accept the privileged requests from a client and Auditor)
+* Scalar DL Auditor
+    * 40051/TCP (Accept the requests from a client and Ledger)
+    * 40052/TCP (Accept the privileged requests from a client and Ledger)
+* Scalar Envoy (Used with Scalar DL Ledger and Scalar DL Auditor)
+    * 50051/TCP (Load balancing for Scalar DL Ledger)
+    * 50052/TCP (Load balancing for Scalar DL Ledger)
+    * 40051/TCP (Load balancing for Scalar DL Auditor)
+    * 40052/TCP (Load balancing for Scalar DL Auditor)
```

Comment:
```suggestion
The default network ports for connecting ScalarDL Ledger, ScalarDL Auditor, and the application (client) by default are as follows. You must allow these connections between each private network.

* **Scalar DL Ledger**
    * **50051/TCP:** Accept requests from an application (client) and ScalarDL Auditor.
    * **50052/TCP:** Accept privileged requests from an application (client) and ScalarDL Auditor.
* **Scalar DL Auditor**
    * **40051/TCP:** Accept requests from an application (client) and ScalarDL Ledger.
    * **40052/TCP:** Accept privileged requests from an application (client) and ScalarDL Ledger.
* **Scalar Envoy** (used with ScalarDL Ledger and ScalarDL Auditor)
    * **50051/TCP:** Accept load balancing requests for ScalarDL Ledger.
    * **50052/TCP:** Accept load balancing requests for ScalarDL Ledger.
    * **40051/TCP:** Accept load balancing requests for ScalarDL Auditor.
    * **40052/TCP:** Accept load balancing requests for ScalarDL Auditor.

Note that, if you change the listening port for ScalarDL in the configuration file (ledger.properties or auditor.properties) from the default, you must allow the connections by using the port that you configured.
```

Reply:
Sorry, my previous description in this doc was very ambiguous...
The concrete request flow is as follows.
```
[Client] ---> [Load balancer] ---> [Kubernetes service (k8s service)] ---> [Scalar Envoy (k8s pod / internal load balancer for gRPC bidirectional streaming)] ---> [ScalarDL (k8s pod)]
```

So, I think the following descriptions are more clear.


```
* **ScalarDL Ledger**
    * **50051/TCP:** Accept requests from an application (client) and ScalarDL Auditor via Scalar Envoy.
    * **50052/TCP:** Accept privileged requests from an application (client) and ScalarDL Auditor via Scalar Envoy.
* **ScalarDL Auditor**
    * **40051/TCP:** Accept requests from an application (client) and ScalarDL Ledger via Scalar Envoy.
    * **40052/TCP:** Accept privileged requests from an application (client) and ScalarDL Ledger via Scalar Envoy.
* **Scalar Envoy** (used with ScalarDL Ledger and ScalarDL Auditor)
    * **50051/TCP:** Accept requests for ScalarDL Ledger from an application (client) and ScalarDL Auditor.
    * **50052/TCP:** Accept privileged requests for ScalarDL Ledger from an application (client) and ScalarDL Auditor.
    * **40051/TCP:** Accept requests for ScalarDL Auditor from an application (client) and ScalarDL Ledger.
    * **40052/TCP:** Accept privileged requests for ScalarDL Auditor from an application (client) and ScalarDL Ledger.
```

What do you think?

Reply:
Thank you for the additional details! I've modified my original suggestion to include the information that you provided👍

```suggestion
The default network ports for connecting ScalarDL Ledger, ScalarDL Auditor, and the application (client) by default are as follows. You must allow these connections between each private network.

* **ScalarDL Ledger**
    * **50051/TCP:** Accept requests from an application (client) and ScalarDL Auditor via Scalar Envoy.
    * **50052/TCP:** Accept privileged requests from an application (client) and ScalarDL Auditor via Scalar Envoy.
* **ScalarDL Auditor**
    * **40051/TCP:** Accept requests from an application (client) and ScalarDL Ledger via Scalar Envoy.
    * **40052/TCP:** Accept privileged requests from an application (client) and ScalarDL Ledger via Scalar Envoy.
* **Scalar Envoy** (used with ScalarDL Ledger and ScalarDL Auditor)
    * **50051/TCP:** Accept requests for ScalarDL Ledger from an application (client) and ScalarDL Auditor.
    * **50052/TCP:** Accept privileged requests for ScalarDL Ledger from an application (client) and ScalarDL Auditor.
    * **40051/TCP:** Accept requests for ScalarDL Auditor from an application (client) and ScalarDL Ledger.
    * **40052/TCP:** Accept privileged requests for ScalarDL Auditor from an application (client) and ScalarDL Ledger.

Note that, if you change the listening port for ScalarDL in the configuration file (ledger.properties or auditor.properties) from the default, you must allow the connections by using the port that you configured.
```